snap/snapcraft.yaml

```
@@ -1,12 +1,87 @@
 name: dcgm
 base: core24
 adopt-info: dcgm
-summary: Snap for NVIDIA DCGM and DCGM exporter
+summary: Snap for NVIDIA DCGM and DCGM-Exporter
 license: Apache-2.0
 contact: solutions-engineering@lists.canonical.com
 description: |
-  This snap includes NVIDIA DCGM and DCGM exporter to provide GPU monitoring via
-  Prometheus metrics. This snap does not include the dcgmproftester.
+  This snap includes NVIDIA DCGM and DCGM-Exporter to manage and monitor NVIDIA GPUs via the CLI or via Prometheus metrics.
+  For instance, the snap can be used to collect the metrics and make Grafana dashboard for data visualization.
+
+  The snap includes the following components:
+  - DCGM: Data Center GPU Manager
+  - DCGM-Exporter: a Prometheus exporter for DCGM metrics
+
+  Please see the links at the bottom of the page for more details about the included components and their purpose.
+
+  **How-To**
+  ---
+
+  **Install the snap**
+
+  ```
+  sudo snap install dcgm
+  ```
+
+  **How to enable metrics collection:**
+
+  ```
+  # Start the DCGM-Exporter service
+  sudo snap start dcgm.dcgm-exporter
+
+  # Get the metrics
+  curl -s localhost:9400/metrics
+  ```
+
+  **Note**: The `DCGM-Exporter` service is disabled by default. If you wish to collect metrics for monitoring,
+  see how to enable the exporter in the section above.
+
+  **How to configure the snap services**
+
+  The DCGM snap provides several configuration options that can be customized through the `snap` CLI.
+  For example:
+
+  ```
+  # Get all the configuration options
+  sudo snap get dcgm
+
+  # Set the NV-Hostengine port
+  sudo snap set dcgm nv-hostengine-port=5577
+
+  # Restart the NV-Hostengine service to apply the changes
+  sudo snap restart dcgm.nv-hostengine
+  ```
+
+  **Reference**
+  ---
+
+  Configurations available:
+
+  - `nv-hostengine-port`: The port on which the NV-Hostengine listens.
+    The default is `5555`.
+  - `dcgm-exporter-address`: The bind address which the DCGM-Exporter exposes for the metrics.
+    The default is `:9400`.
+  - `dcgm-exporter-metrics-file`: The name of the custom CSV metrics file can be provided (only the name, not the path).
+    The file should be placed in the `/var/snap/dcgm/common/` directory.
+    The default metrics are located in `/snap/dcgm/current/etc/dcgm-exporter/default-counters.csv`.
+    Please refer to the DCGM-Exporter repository link at the bottom of the page for more information on the CSV file format.
+
+  This snap does not include the `dcgmproftester`, which is a performance testing tool, to limit the size of the snap.
```

I would readd the "Limitations" header otherwise this sentence feels out of place. Also we can generally assume the users know what the components are:
```suggestion
  The DCGM snap does not currently include the performance testing tool `dcgmproftester` in order to limit the size of the snap.
```